### PR TITLE
[improvement] update default firewall to restrict API server access and fix nodeport range

### PR DESCRIPTION
**Pull request by eljohnson92**
**What this PR does / why we need it**:
This PR tweaks firewall rules to be more accurate restrictive in the following ways:
- [x] by default the API server port is restricted only to the nodebalancer range since it isn't needed to talk be talked to directly
- [x] the nodeport range is expanded from 30000-30100 to the nodeport default range of 30000-32767
- [x] konnectivity port rules are only enabled in the konnectivity flavor
- [x] dns based loadbalancing flavors have specific API server rules to allow direct connection to the API server

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/flavors/rke2/dns-loadbalancing/kustomization.yaml [hunk 1]
```diff
@@ -20,3 +20,55 @@ patches:
...
+            label: inbound-api-server
+            description: accept all api server and konnectivity related traffic from nodebalancers
+          - action: ACCEPT
+            addresses:
+              ipv4:
+                - 192.168.255.0/24
+            ports: 30000-32767
+            protocol: TCP
+            label: accept-NodeBalancer
+            description: accept traffic from the entire NodeBalancer CIDR to the NodePort service range
```
Comment (komer3) [inline templates/flavors/rke2/dns-loadbalancing/kustomization.yaml:74]: nit: new line at the end of file

File: templates/flavors/k3s/dns-loadbalancing/kustomization.yaml [hunk 2]
```diff
@@ -20,3 +20,55 @@ patches:
...
+            label: inbound-api-server
+            description: accept all api server and konnectivity related traffic from nodebalancers
+          - action: ACCEPT
+            addresses:
+              ipv4:
+                - 192.168.255.0/24
+            ports: 30000-32767
+            protocol: TCP
+            label: accept-NodeBalancer
+            description: accept traffic from the entire NodeBalancer CIDR to the NodePort service range
```
Comment (komer3) [inline templates/flavors/k3s/dns-loadbalancing/kustomization.yaml:74]: nit: new line at the end of file

**Review by komer3 — COMMENTED** [review]
apart from the small nit comments. LGTM!

File: templates/flavors/k3s/dns-loadbalancing/kustomization.yaml [hunk 3]
```diff
@@ -20,3 +20,55 @@ patches:
...
+              ipv4:
+                - "10.0.0.0/8"
+            description: accept all icmp traffic within the vpc
+          - action: ACCEPT
+            addresses:
+              ipv4:
+                - 0.0.0.0/0
+              ipv6:
+                - ::/0
+            ports: "${APISERVER_PORT:=6443}"
```
Comment (rahulait) [inline templates/flavors/k3s/dns-loadbalancing/kustomization.yaml:63]: Should we have konnectivity port here based on description?

File: templates/flavors/kubeadm/dns-loadbalancing/kustomization.yaml [hunk 4]
```diff
@@ -20,3 +20,55 @@ patches:
...
+              ipv4:
+                - "10.0.0.0/8"
+            description: accept all icmp traffic within the vpc
+          - action: ACCEPT
+            addresses:
+              ipv4:
+                - 0.0.0.0/0
+              ipv6:
+                - ::/0
+            ports: "${APISERVER_PORT:=6443}}"
```
Comment (rahulait) [inline templates/flavors/kubeadm/dns-loadbalancing/kustomization.yaml:63]: for kubeadm, do we need to add konnectivity port here?

Comment (rahulait) [reply templates/flavors/kubeadm/dns-loadbalancing/kustomization.yaml:63]: If we expect dns-loadbalancing flavor to not have konnectivity support, then we should be good here

Comment (eljohnson92) [reply templates/flavors/kubeadm/dns-loadbalancing/kustomization.yaml:63]: I don't think we have any flavor that does konnectivity + DNS loadbalancing right now, if we added one we could include this port information as well

File: templates/flavors/kubeadm/dns-loadbalancing/kustomization.yaml [hunk 5]
```diff
@@ -20,3 +20,55 @@ patches:
...
+          - action: ACCEPT
+            addresses:
+              ipv4:
+                - 0.0.0.0/0
+              ipv6:
+                - ::/0
+            ports: "${APISERVER_PORT:=6443}"
+            protocol: TCP
+            label: inbound-api-server
+            description: accept all api server and konnectivity related traffic from nodebalancers
```
Comment (rahulait) [inline templates/flavors/kubeadm/dns-loadbalancing/kustomization.yaml:66]:
```suggestion
            description: accept all api server related traffic from nodebalancers
```

**Review by rahulait — APPROVED** [review]

**Review by komer3 — APPROVED** [review]

**Pull request merged**